docs/generate_py_docs.py

```diff
@@ -0,0 +1,170 @@
+            # add new line and indent for the cells inside a table
+            if 'class' not in tag.attrs:
+                tag.insert_before('\n      ')
+
+    # substitute double backticks with a single backtick
+    for tag in html_elements.select('tt'):
+        tag.insert_before('`')
+        tag.insert_after('`')
+
+    # make the option's text inside the option list bold
+    for tag in html_elements.select('span'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) == 'option':
+                tag.insert_before('<b>')
+                tag.insert_after('</b>')
+
+    # keep bold text bold
+    for tag in html_elements.select('strong'):
+        tag.insert_before('<b>')
+        tag.insert_after('</b>')
+
+    # add new lines after the paragraph tag
+    for tag in html_elements.select('p'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) == 'first':
+                tag.insert_before(' ')
+            else:
+                tag.insert_before('\n')
+        else:
+            tag.insert_before('\n\n')
+
+    # format elements in the `Option:` section by adding new line and indent
+    for tag in html_elements.select('kbd'):
+        tag.insert_before('\n   ')
+
+    # add indent to the description list
+    for tag in html_elements.select('dt'):
+        tag.string = '\n' + indent(tag.text.strip(), '   ')
+
+    # add indent to the nested description list
+    for tag in html_elements.select('dd'):
+        tag.string = '\n' + indent(tag.text, '      ')
+
+    # add indent and bullet points to the bulleted list
+    for tag in html_elements.select('li'):
+        if tag.parent.parent.name == 'li':  # deal with nested lists
+            tag.insert_before('\n   - ')
+        else:
+            tag.insert_before('\n* ')
+
+    # add new line before included NOTE class of rst documentation, example: github_api_access.rst
+    for tag in html_elements.select('div'):
+        if 'class' in tag.attrs:
+            if 'admonition note' in ' '.join(tag.attrs['class']):
+                tag.insert_before('\n')
+
+    # format code examples
+    for tag in html_elements.select('pre'):
+        if 'class' in tag.attrs:
+            if ' '.join(tag.attrs['class']) in ['code shell literal-block', 'code awk literal-block']:
+                tag.string = '<b>' + indent(tag.text.rstrip('\n'), '  ') + '</b>'
+            elif 'literal-block' in ' '.join(tag.attrs['class']):
+                tag.string = '\n<dim>' + indent(tag.text, '  ') + '</dim>'
+
+    # substitute color classes with respective ANSI codes
+    for tag in html_elements.select('span'):
+        if 'class' in tag.attrs:
+            color = ' '.join(tag.attrs['class']).replace('gray', 'dim')
+            if color in ['green', 'yellow', 'red', 'dim']:
+                tag.insert_before(f'<{color}>')
+                tag.insert_after(f'</{color}>')
+
+    # build plain text output out of the previously formatted string elements
```

Comment:
Hmm, without this step there won't be any output docs text because previous steps just modified the html docs but they did not build actual plain text docs, maybe:

```
# build python docs string out of the previously formatted html tags
```
is better?

Reply:
Aww okay 🤦🏻 yes, pls change